cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1867,7 +1867,7 @@ auto convert_table_to_parquet_data(table_input_metadata& table_meta,
         ck                   = {};
         ck.col_desc          = col_desc.device_ptr() + c;
         ck.col_desc_id       = c;
-        ck.fragments         = &row_group_fragments.device_view()[c][f];
+        ck.fragments         = row_group_fragments.device_view()[c].data() + f;
```

Comment:
We used to call `device_span::operator[]` here, which is a device function.
Compiler is okay with this, but does not sound like something we should do.

Reply:
Wonder why that does not throw compiler error?

Reply:
Probably because the function is `constexpr`. I think it would fail once we turn off the "treat constexpr functions and `__host__ __device__`".